src/Adapter/SDL/Render/Engine.php

```diff
@@ -127,15 +122,27 @@ public function start(Presentation\SlideShow $slideShow, Graphic\Drawer $drawer)
         echo PHP_EOL;
     }
 
-    private function loadingScreen(DbgTextRenderer $textRenderer, Presentation\Progress $progress): void
+    private function preload(DbgTextRenderer $textRenderer, Presentation\SlideShow $slideShow, Graphic\Drawer $drawer): void
     {
+        reload:
+
+        $w = $h = 0;
+        \SDL_GetRendererOutputSize($this->renderer, $w, $h);
+        $this->screen = $this->screen->resized(
+            Geometry\Size::fromDimensions($w, $h)
+        );
+
+        $progress = $slideShow->preload($this->screen, $drawer);
+
         $size = (int) ($this->screen->safeArea()->size()->height() / 30);
         $event = new \SDL_Event();
         $current = $progress->advance();
         while ($current < $progress->count()) {
             // Inputs polling
             while (sdl_pollevent($event) !== 0) {
-                // No op
+                if ($event->type === SDL_WINDOWEVENT && $event->window->event === SDL_WINDOWEVENT_RESIZED) {
+                    goto reload; // Avoid nesting function calls
```

Comment:
We could imagine a global `while(! $isCompletelyLoaded)` loop, but this `goto` is pretty straightforward and easy to understand/maintain 👍